modules/apps/static/portal-file-install/portal-file-install-test/src/testIntegration/java/com/liferay/portal/file/install/deploy/test/FileInstallConfigTest.java

```diff
@@ -307,6 +307,46 @@ public void testFactoryConfigurationWithUnderline() throws Exception {
 		_testFactoryConfiguration(CharPool.UNDERLINE);
 	}
 
+	@Test
+	public void testMultipleFactoryConfigurations() throws Exception {
+		_testFactoryConfiguration(CharPool.TILDE);
+
+		String factoryConfigurationName = StringBundler.concat(
+			StringUtil.randomId(), CharPool.DASH, StringUtil.randomId());
+
+		String factoryPid = _CONFIGURATION_PID_PREFIX.concat(
+			".testFactoryConfiguration");
+
+		Path factoryConfigurationPath = Paths.get(
+			PropsValues.MODULE_FRAMEWORK_CONFIGS_DIR,
+			StringBundler.concat(
+				factoryPid, CharPool.TILDE, factoryConfigurationName,
+				".config"));
+
+		_createFactoryConfiguration(
```

Comment:
The reason why you have to sleep is because the latch never got resolved. Its basically a lock that we learned in the architect classes. Looks up https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CountDownLatch.html
and ConfigurationTestUtil.java (some test examples).

Most likely the latch is .awaiting on the old condition since we are re-using the same configuration path, but since its true it instantly goes? I would recommend refactoring  _testFactoryConfig instead to run multiple times since we should be re-using it instead